packages/headless/src/api/search/date/date-format.ts

```diff
@@ -2,11 +2,26 @@ import dayjs from 'dayjs';
 import customParseFormat from 'dayjs/plugin/customParseFormat';
 dayjs.extend(customParseFormat);
 
+const API_DATE_FORMAT = 'YYYY/MM/DD@HH:mm:ss';
+
 export function formatDateForSearchApi(date: dayjs.Dayjs) {
-  const DATE_FORMAT = 'YYYY/MM/DD@HH:mm:ss';
-  return date.format(DATE_FORMAT);
+  if (!date.isValid()) {
+    throw new Error('Date object is invalid.');
+  }
+
+  if (date.isBefore('1401-01-01')) {
+    throw new Error(
+      'Date is before year 1401, which is unsupported by the index.'
+    );
+  }
+
+  return date.format(API_DATE_FORMAT);
 }
 
 export function isSearchApiDate(date: string) {
-  return formatDateForSearchApi(dayjs(date)) === date;
+  try {
+    return formatDateForSearchApi(dayjs(date)) === date;
+  } catch (error) {
+    return false;
+  }
```

Comment:
So, there are different schools of thought on throwing errors. There are pros to throwing from a deep function: less duplication, and a lower chance of missing something.

When a deep function throws though, it forces every function using it to wrap it's code in a try catch, like what is happening here. If one instance is added and the person forgets it can throw (i.e. forgets to catch), this can lead to a bug and maintenance case at run-time though.

Another approach is to view validation and formatting as separate steps. When we are at the formatting stage, we do not want to be worrying about validating.

Personally, I would prefer if we validated at the entry points (or close to them) instead of deeper inside a formatting function.

Headless provides a utility for creating dates, so adding a [minimum date limit here](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/controllers/core/facets/range-facet/date-facet/date-range.ts#L95) seems like a perfect place.

Going one step further, we could also [validate at the controller level](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/controllers/core/facets/range-facet/date-facet/headless-core-date-facet.ts#L159), and the action level (as in this PR).

By moving validation to the edges, we would also remove the need for `catch` statements,

Reply:
We now validate and format in separate methods, in separate steps. We just need to handle it in 2 places in Headless, I added a comment for each